cmake/setup_custom_targets.cmake

```diff
@@ -60,13 +60,19 @@ FOREACH(_component compat_files documentation examples parameter_gui)
     _add_custom_target(${_component})
   ELSE()
     STRING(TOUPPER ${_component} _componentuppercase)
+
+    SET(_first_message "Error: Could not ${_description_string} disabled component ${_component}.")
+    DECORATE_WITH_STARS(${_first_message} _decorated_first_message)
+    SET(_second_message "Please reconfigure with -DDEAL_II_COMPONENT_${_componentuppercase}=ON")
+    DECORATE_WITH_STARS(${_second_message} _decorated_second_message)
+
     ADD_CUSTOM_TARGET(${_component}
       COMMAND
            ${CMAKE_COMMAND} -E echo ''
         && ${CMAKE_COMMAND} -E echo ''
         && ${CMAKE_COMMAND} -E echo '***************************************************************************'
-        && ${CMAKE_COMMAND} -E echo "**  Error: Could not ${_description_string} disabled component \"${_component}\"."
-        && ${CMAKE_COMMAND} -E echo "**  Please reconfigure with -DDEAL_II_COMPONENT_${_componentuppercase}=ON"
+        && ${CMAKE_COMMAND} -E echo "${_decorated_first_message}"
+        && ${CMAKE_COMMAND} -E echo "${_decorated_second_message}"
```

Comment:
i agree that readability for developers suffers. But this is the way to make **consistent output** as in this case `_description_string` and `_component` are variables. So whatever formatting we decide on, it would be hard to do it consistently by hand. On the positive side for users, with macros messages will be formatted consistently and nicely.

So I personally :+1: this macro, but let's see if others want to contribute to the discussion.